core/build/build.odin

```diff
@@ -0,0 +1,304 @@
+		
+				case arg == "-cwd-workspace":
+					build_project = false
+					settings.dev_opts.flags += {.Cwd_Workspace}
+		
+				case arg == "-cwd-out":
+					build_project = false
+					settings.dev_opts.flags += {.Cwd_Out}
+		
+				case arg == "-install":
+					is_install = true
+		
+				case strings.has_prefix(arg, "-cwd"):
+					build_project = false
+					log.warnf("Flag %s not implemented", arg)
+		
+				case strings.has_prefix(arg, "-launch-args"):
+					build_project = false
+					log.warnf("Flag %s not implemented", arg)
+		
+				case strings.has_prefix(arg, "-include-build-system"):
+					build_project = false
+					settings.dev_opts.flags += {.Include_Build_System}
+					log.warnf("Flag %s not implemented\n", arg)
+		
+				case: log.errorf("Invalid flag %s", arg)
+			}
+		} else {
+			if config_name != "" {
+				log.errorf("Config already set to %s and cannot be re-assigned to %s. Cannot have 2 configurations built with the same command (yet).", config_name, arg)
+				return
+			}
+			config_name = arg // Todo(Dragos): Remove this variable
+			settings.target_name = config_name
+		}
+	}
+
+	// Note(Dragos): I do not like this approach
+	if display_help {
+		settings.command_type = .Display_Help
+	} else if build_project {
+		settings.command_type = .Install if is_install else .Build
+	} else {
+		settings.command_type = .Dev
+	}
+	return true
+}
+
+add_target :: proc(project: ^Project, target: ^Target, loc := #caller_location) {
+	append(&project.targets, target)
+	target.project = project
+	file_dir := filepath.dir(loc.file_path, context.temp_allocator)
+	root_dir := filepath.join({file_dir, ".."}, context.temp_allocator)
+	target.root_dir = filepath.clean(root_dir) // This effectively makes the build system dependent on being a subfolder of the main project root
+}
+
+target_config :: proc(target: ^Target, settings: Settings) -> Config {
+	if config, cached := target.cached_config.?; cached do return config
+	target.cached_config = target.project->configure_target_proc(target, settings)
+	#no_type_assert return target.cached_config.?
+}
+
+Add_Dependency_Error :: enum {
+	None,
+	Circular_Dependency,
+}
+
+find_dependency :: proc(target: ^Target, depend: ^Target) -> bool {
+	// Note(Dragos): is this logic correct?
+	if target == depend do return true
+	for d in target.depends do return find_dependency(d, depend)
+	return false
+}
+
+add_dependency :: proc(target: ^Target, depend: ^Target) -> (err: Add_Dependency_Error) {
+	if find_dependency(target, depend) do return // dependency already present	 
+	if find_dependency(depend, target) {
+		log.errorf("Circular dependency detected for [%s - %s] and [%s - %s]", target.project.name, target.name, depend.project.name, depend.name)
+		return .Circular_Dependency
+	}
+	append(&target.depends, depend)
+	return
+}
+
+run :: proc(main_project: ^Project, opts: Settings) {
+	opts := opts
+	if opts.target_name == "" do opts.target_name = opts.default_target_name
```

Comment:
I strongly suggest against using `do` in `core` packages. Though in this case it may be fine though because it's going to be compiled from a build script where people wouldn't put `-disallow-do` compiler flag. Maybe still worth removing it for non-typical usecases of this package.